internal/cli/universal_login_customize.go

```diff
@@ -237,3 +252,92 @@ func fetchPromptCustomTextWithDefaults(
 		CustomText: brandingTextTranslations,
 	}, nil
 }
+
+func startWebSocketServer(
+	ctx context.Context,
+	api *auth0.API,
+	display *display.Renderer,
+	brandingData *universalLoginBrandingData,
+) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return err
+	}
+	defer listener.Close()
+
+	handler := &webSocketHandler{
+		display:      display,
+		api:          api,
+		shutdown:     cancel,
+		brandingData: brandingData,
+	}
+
+	server := &http.Server{
+		Handler:      handler,
+		ReadTimeout:  time.Minute * 10,
+		WriteTimeout: time.Minute * 10,
+	}
+
+	errChan := make(chan error, 1)
+	go func() {
+		errChan <- server.Serve(listener)
+	}()
+
+	openWebAppInBrowser(display, listener.Addr())
+
+	select {
+	case err := <-errChan:
+		return err
+	case <-ctx.Done():
+		return server.Close()
+	}
+}
+
+func openWebAppInBrowser(display *display.Renderer, addr net.Addr) {
+	port := addr.(*net.TCPAddr).Port
+	webAppURLWithPort := fmt.Sprintf("%s?ws_port=%d", webAppURL, port)
+
+	display.Infof("Perform your changes within the UI: %q", webAppURLWithPort)
```

Comment:
```suggestion
	display.Infof("Perform your changes within the editor: %q", webAppURLWithPort)
```
A minor nit, UI is from our PoV.